src/addrman.cpp

```diff
@@ -721,13 +721,18 @@ std::pair<CAddress, NodeSeconds> AddrManImpl::Select_(bool new_only, std::option
     size_t new_count = nNew;
     size_t tried_count = nTried;
 
-    if (network.has_value()) {
-        auto it = m_network_counts.find(*network);
-        if (it == m_network_counts.end()) return {};
-
-        auto counts = it->second;
-        new_count = counts.n_new;
-        tried_count = counts.n_tried;
+    if (!networks.empty()) {
```

Comment:
About:
> I tried to test the performance impact on the "normal" case of an addrman that only has addrs for reachable nets:
If I change the AddrManSelect benchmark to use addrman.Select(false, g_reachable_nets.All());  the performance goes down by a factor of ~5 for me (from ~160ns/op to ~770 ns/op). Do you see a similar effect?

In fact, I just realized that the performance issue is related to `g_reachable_nets.All()`. With the following diff I realized the performance is not relevantly affected. cc: @mzumsande 

```diff
diff --git a/src/bench/addrman.cpp b/src/bench/addrman.cpp
index ee4070b07a..b980c7dec1 100644
--- a/src/bench/addrman.cpp
+++ b/src/bench/addrman.cpp
@@ -89,9 +89,10 @@ static void AddrManSelect(benchmark::Bench& bench)
     AddrMan addrman{EMPTY_NETGROUPMAN, /*deterministic=*/false, ADDRMAN_CONSISTENCY_CHECK_RATIO};
 
     FillAddrMan(addrman);
+    auto nets{g_reachable_nets.All()};
 
     bench.run([&] {
-        const auto& address = addrman.Select();
+        const auto& address = addrman.Select(false, nets);
         assert(address.first.GetPort() > 0);
     });
 }
```

Reply:
> Based on the above, I think that it is best to go back to 7edb07ca80 and only fix the Select() argument.

Since it is simpler and does not impact the performance, I just addressed this suggestion. Thanks.